common/ostream.h

```diff
@@ -62,6 +62,35 @@ void PrintTo(T* p, std::ostream* out) {
   }
 }
 
+// True if T can be used with `<<` and an `llvm::raw_ostream`.
+template <typename T>
+static constexpr bool HasRawOstream = Requires<const T, llvm::raw_ostream>(
```

Comment:
The name `HasRawOstream` isn't working well for me. How about something like `IsStreamableToRawOstream`? It's a bit long but seems hard to misinterpret.